src/structures/Template.ts

```diff
@@ -0,0 +1,136 @@
+import {
+	APITemplate,
+	ChannelType,
+	GuildDefaultMessageNotifications,
+	GuildExplicitContentFilter,
+	GuildSystemChannelFlags,
+	GuildVerificationLevel,
+	OverwriteType,
+	Permissions,
+	Snowflake,
+} from 'discord-api-types';
+import { Client } from '../clients';
+import { User } from './User';
+
+interface TemplateSerializedSourceGuild {
+	description?: string;
+	preferredLocale: string;
+	iconHash?: string;
+	name: string;
+	region?: string;
+	verificationLevel?: GuildVerificationLevel;
+	defaultMessageNotifications?: GuildDefaultMessageNotifications;
+	explicitContentFilter?: GuildExplicitContentFilter;
+	roles?: GuildCreateRole[];
+	channels?: GuildCreatePartialChannel[];
+	afkChannelId?: number | Snowflake;
+	afkTimeout?: number;
+	systemChannelId?: number | Snowflake;
+	systemChannelFlags?: GuildSystemChannelFlags;
+}
+
+interface GuildCreateRole {
+	name?: string;
+	permissions?: Permissions;
+	color?: number;
+	hoist?: boolean;
+	mentionable?: boolean;
+	id: number | string;
+}
+
+interface GuildCreatePartialChannel {
+	type?: ChannelType;
+	topic?: string;
+	nsfw?: boolean;
+	bitrate?: number;
+	userLimit?: number;
+	rateLimitPerUser?: number;
+	name: string;
+	id?: number | string;
+	parentId?: number | string;
+	permissionOverwrites?: GuildCreateOverwrite[];
+}
+
+interface GuildCreateOverwrite {
+	allow: Permissions;
+	deny: Permissions;
+	type: OverwriteType;
+	id: number | string;
+}
+
+export class Template {
+	public code!: string;
+	public name!: string;
+	public description?: string;
+	public usageCount!: number;
+	public creatorId!: Snowflake;
+	public creator!: User;
+	public createdAt!: Date;
+	public updatedAt!: Date;
+	public sourceGuildId!: Snowflake;
+	public serializedSourceGuild!: TemplateSerializedSourceGuild;
+	public isDirty!: boolean;
+
+	public constructor(public readonly client: Client, data: APITemplate) {
+		this.$patch(data);
+	}
+
+	public $patch(data: APITemplate): void {
+		if (data.description) {
+			this.description = data.description;
+		}
+
+		this.code = data.code;
+		this.name = data.name;
+		this.usageCount = data.usage_count;
+		this.creatorId = data.creator_id;
+		this.creator = new User(this.client, data.creator);
+		this.createdAt = new Date(data.created_at);
+		this.updatedAt = new Date(data.updated_at);
+		this.sourceGuildId = data.source_guild_id;
+		this.isDirty = !!data.is_dirty;
```

Comment:
```suggestion
		this.isDirty = data.is_dirty === null ? undefined : data.is_dirty;
```
(also update the type)